sympy/physics/control/lti.py

```diff
@@ -0,0 +1,655 @@
+        The numerator polynomial of the transfer function.
+    den : Expr, Number
+        The denominator polynomial of the transfer function.
+    var : Symbol
+        Variable used to anchor the polynomials of the transfer function.
+
+    Raises
+    ======
+
+    TypeError
+        When ``var`` is not a Symbol or when ``num`` or ``den`` is not
+        a number or expression/polynomial. Also, when ``num`` or ``den`` has
+        a time delay term.
+    ValueError
+        When ``den`` is zero.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import s, p, a
+    >>> from sympy.physics.control.lti import TransferFunction
+    >>> tf1 = TransferFunction(s + a, s**2 + s + 1, s)
+    >>> tf1
+    TransferFunction(a + s, s**2 + s + 1, s)
+    >>> tf1.num
+    a + s
+    >>> tf1.den
+    s**2 + s + 1
+    >>> tf1.var
+    s
+
+    Using different polynomial variables.
+
+    >>> tf2 = TransferFunction(a*p**3 - a*p**2 + s*p, p + a**2, p)
+    >>> tf2
+    TransferFunction(a*p**3 - a*p**2 + p*s, a**2 + p, p)
+    >>> tf3 = TransferFunction((p + 3)*(p - 1), (p - 1)*(p + 5), p)
+    >>> tf3
+    TransferFunction((p - 1)*(p + 3), (p - 1)*(p + 5), p)
+
+    Using ``-`` operator in front of a transfer function to negate it.
+
+    >>> tf4 = TransferFunction(-a + s, p**2 + s, p)
+    >>> -tf4
+    TransferFunction(a - s, p**2 + s, p)
+    >>> tf5 = TransferFunction(s**4 - 2*s**3 + 5*s + 4, s + 4, s)
+    >>> -tf5
+    TransferFunction(-s**4 + 2*s**3 - 5*s - 4, s + 4, s)
+
+    Using Float or Integer (or other constants) as numerator and denominator.
+
+    >>> tf6 = TransferFunction(1/2, 4, s)
+    >>> tf6.num
+    0.500000000000000
+    >>> tf6.den
+    4
+    >>> tf6.var
+    s
+
+    Using ``**`` operator to take the power of a transfer function.
+
+    >>> tf7 = TransferFunction(s + a, s - a, s)
+    >>> tf7**3
+    TransferFunction((a + s)**3, (-a + s)**3, s)
+    >>> tf7**0
+    TransferFunction(1, 1, s)
+    >>> tf8 = TransferFunction(p + 4, p - 3, p)
+    >>> tf8**-1
+    TransferFunction(p - 3, p + 4, p)
+
+    References
+    ==========
+
+    .. [1] http://www.cds.caltech.edu/~murray/amwiki/index.php?title=First_Edition
+
+    """
+    def __new__(cls, num, den, var):
+        num, den = _sympify(num), _sympify(den)
+
+        if not isinstance(var, Symbol):
+            raise TypeError("Variable input must be a Symbol.")
+        if den == 0:
+            raise ValueError("TransferFunction can't have a zero denominator.")
+
+        if (((isinstance(num, Expr) and num.has(Symbol) and not num.has(exp)) or num.is_number) and
+            ((isinstance(den, Expr) and den.has(Symbol) and not den.has(exp)) or den.is_number)):
+                obj = super(TransferFunction, cls).__new__(cls, num, den, var)
+                obj._num = num
+                obj._den = den
+                obj._var = var
+                return obj
+        else:
+            raise TypeError("Unsupported type for numerator or denominator of TransferFunction.")
```

Comment:
I think you need to raise a ValueError if the numerator or denominators are expressions that aren't polynomials in var.